Upgrading to an addon [Tested & Released]

Nomas-X opened the pull request:
The following PR includes the following key features:

1. hemtt support and integration.
2. CBA settings to allow the disabling and enabling of certain features [i.e. telling victim].
3. Github actions for handling packaging, signing and release of the mod.
4. Added macros that are used in this switch to a mod / addon.
5. Minor code improvements because hemtt yelled at me.
6. Major rework of the file structure.
7. Renamed functions to reflect new name and get rid of "punishment".

To do if PR is merged:
1. Update repo name to `Friendly-Fire-Phantom-Protocol`.

Nomas-X committed:
Initial Commit

- Removing all data to rebuild mod.

Nomas-X committed:
Adding Github Actions

- Added auto-packging, signing and release actions.
- Added sqf-validator action.

Nomas-X committed:
Setting up hemtt & macros

- Finished setting up hemtt
- Added CBA macros.
- Added tools and scripts.

Nomas-X committed:
Creation of addons

- Created the core addon which contains all the functions.
- Created the main addon which contains the macros and more.

Nomas-X committed:
Macro fixes 1

- Setting up and fixing some macros since we are not using ACE.

Nomas-X committed:
Fixing Macros 2

- Macros should all be fixed and usable now.

Nomas-X committed:
Image Path fixes

- Fixed image paths in custom hint

Nomas-X committed:
Update mission.sqm

- Added a proper mission for testing.

Nomas-X committed:
Update README.md

Nomas-X committed:
Create steamLogo.png

- Placeholder picture to be used for steam release?

Nomas-X committed:
Create mod.cpp

- Added the mod presentation file

Nomas-X committed:
Added placeholder logos

- Added placeholder logos for mod presentation

Nomas-X committed:
Updated mod name

- Updated mod name to Friendly Fire Phantom Protocol

Nomas-X committed:
Update release.yml

- Removed incorrect item ID. Must be updated.

Nomas-X committed:
Combined into 1 addon

- Got rid of the 2 addon set up and instead just combined it into a single addon as there will be no plans to add future addons.

Nomas-X committed:
Reorganized & Renamed Functions

- Renamed and reorganized functions to remove the word Punishment as no actual punishment is present in the mod. Additionally makes function names easier to read.

Nomas-X committed:
Update README.md

Nomas-X committed:
Update config.cpp

- Added Frost'sBite second name that was mentioned in functions.

Nomas-X committed:
Updated images

- Removed `TM` from the images and did some minor improvement.

Nomas-X committed:
Minor comment and string changes

- Removed the word Punishment from some places and replaced with the new name.

Nomas-X committed:
Update release.yml

- Updated name

Nomas-X committed:
Update release.yml

- Added workshop ID

CalebSerafin commented on include/A3/Functions_F_Exp_A/EGSpectatorCommonDefines.inc:
For `EGSpectatorCommonDefines.inc` I'm pretty sure you can just #include that from the base game files, you don't need to copy it.
Also, I don't see anything that includes `EGSpectatorCommonDefines.inc`?

Nomas-X replied:
The include folder is simply there for using hemtt launch while not in a P Drive. It is not copied into the mod on build or release if that is what you mean.

CalebSerafin commented on addons/main/XEH/XEH_preInit.sqf:
```diff
@@ -0,0 +1,60 @@
+	false									//    _needRestart - Setting will be marked as needing mission restart after being changed. (optional, default false) <BOOL>
+] call CBA_fnc_addSetting;
+*/
+
```
You can make the CBA optional by doing an early exit: `if (isNil "CBA_fnc_addSetting") exitWith { }`
In this case, no CBA options are displayed if CBA is not loaded. There is also a macro way to do this.
Just test because it's being a while since I wrote SQF.

Nomas-X replied:
Can add that to the to do list, is it required for merge?

CalebSerafin commented on addons/main/functions/utility/fn_log.sqf:
```diff
@@ -18,25 +18,25 @@ private _toServer = param [3, !(hasInterface && _level > 1)];
 switch (_level) do {
 	case 1: {
-		_logLine = format ["%1: [Antistasi] | ERROR | %2 | %3", servertime, _file, _message];
+		_logLine = format ["%1: [Antistasi] | ERROR | %2 | %3", serverTime, _file, _message];
```
Ah whoops, could you do a solid and replace [Antistasi] with something like [FF Phantom].

Nomas-X replied:
Will do

CalebSerafin commented on addons/main/script_macros.hpp:
```diff
@@ -0,0 +1,6 @@
+// CBA Settings
+#define ADDON_NAME QUOTE(PREFIX_BEAUTIFIED COMPONENT_BEAUTIFIED)
+#define SET(var1) TRIPLES(ADDON,SET,var1)
+#define QSET(var1) QUOTE(SET(var1))
```
Hmm, should look at extracting the required macro from CBA's script_macros so that we don't need a dependency just for that.

Nomas-X replied:
Can probably do though with CBA settings being rather useful and potential plans down the line to use CBA custom events instead of remoteExec to reduce network traffic I suggest we keep CBA. Working with it always present is just less headache than trying to make both solutions work imo.

CalebSerafin commented on include/A3/ui_f/hpp/defineCommonGrids.inc:
I'm pretty sure this can also be #include from the base game files, you don't need to copy it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/A3/ui_f/hpp/defineDIKCodes.inc:
I'm pretty sure this can also be #include from the base game files, you don't need to copy it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/A3/ui_f/hpp/defineResincl.inc:
I'm pretty sure this can also be #include from the base game files, you don't need to copy it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/A3/ui_f/hpp/defineResinclDesign.inc:
I'm pretty sure this can also be #include from the base game files, you don't need to copy it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/A3/ui_f_exp_a/UI/defineResinclDesign.inc:
I'm pretty sure this can also be #include from the base game files, you don't need to copy it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/x/cba/addons/main/script_macros_common.hpp:
Okay, this one is kind of long. But if you place a dependency on CBA, then you don't need to have the file in your addon and can just #include it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin commented on include/x/cba/addons/xeh/script_xeh.hpp:
if you place a dependency on CBA, then you don't need to have the file in your addon and can just #include it.

Nomas-X replied:
Same as previous. include file is simply there to allow quick testing by launch command when not using a P Drive.

CalebSerafin reviewed (changes requested):
I haven't looked at the actions and sqf validator yet.
but I've made comments on the included .hpp files which I'm pretty sure don't need to be copied into this addon and can just be referenced.

CalebSerafin reviewed (approved):
LGTM

Nomas-X committed:
Update XEH_preInit.sqf

- Corrected typo

Nomas-X committed:
Update `release..yml`

- removing `v` from the changelog link as we do not use it in version releases, we use only numbers.
